libraries/AP_GPS/AP_GPS_NMEA.cpp

```diff
@@ -306,7 +306,10 @@ bool AP_GPS_NMEA::_term_complete()
                     make_gps_time(_new_date, _new_time * 10);
                     state.last_gps_time_ms = now;
                     // To-Do: add support for proper reporting of 2D and 3D fix
-                    state.status           = AP_GPS::GPS_OK_FIX_3D;
+                    if (_gga_gps_status > 0) //use GGA quality indicator if available
```

Comment:
This should be compared against GPS_NO_GPS rather then 0. As currently written it will always step into here, as you've gotten the value of GPS_NO_FIX wrong (it's 1 not 0). Always use the enum name.

Reply:
It appears that GGA is actually a required message, and we could handle the status information in just the one switch in GGA handler, which would be cleaner from a code perspective and remove the need to do this if/comparison stuff up here.